httputil/retry.go

```diff
@@ -82,16 +83,34 @@ func ShouldRetryError(attempt *retry.Attempt, err error) bool {
 				logger.Debugf("Retrying because of: %s", opErr)
 				return true
 			}
+			// FIXME: code below is not (unit) tested and
+			// it is unclear if we need it with the new
+			// opErr.Temporary() "if" below
 			if opErr.Op == "dial" {
 				logger.Debugf("Retrying because of: %#v (syscall error: %#v)", opErr, syscallErr.Err)
 				return true
 			}
 			logger.Debugf("Encountered syscall error: %#v", syscallErr)
 		}
-		if opNetErr, ok := opErr.Err.(net.Error); ok {
-			// TODO: some DNS errors? just log for now
-			logger.Debugf("Not retrying: %#v", opNetErr)
+
+		// If we are unable to talk to a DNS go1.9+ will set
+		// opErr.IsTemporary - we also support go1.6 so we need to
+		// add a workaround here. This block can go away once we
+		// use go1.9+ only.
+		if dnsErr, ok := opErr.Err.(*net.DNSError); ok {
+			// The horror, the horror
+			if strings.Contains(dnsErr.Err, "connection refused") {
```

Comment:
I think the Cgo resolver may actually return a different error message than the purge Go one. IIRC pure Go one is preferred because of goroutines having lower overhead than the actual glibc call blocking the whole thread. That `connection refused` is from the Go resolver. Maybe it'd be worth forcing the resolver in the spread test and checking whether we cover the Cgo case too.